src/electionguard/ballot.cpp

```diff
@@ -619,6 +629,11 @@ namespace electionguard
         return pimpl->proof.get();
     }
 
+    unique_ptr<HashedElGamalCiphertext> CiphertextBallotContest::getHashedElGamalCiphertext() const
+    {
+        return pimpl->hashedElGamal.get()->clone();
```

Comment:
This is a question, not a request to change... do we need to clone this here?

Reply:
This is an excellent question. I originally didn't have it as a clone. However, when I was using it in some of the tests that grabbed the hashed elgamal object out of the CiphertextBallotContest they were blowing up because the object would go away while being used. I added the clone and the problem went away. Not sure this is a great explanation because I am not sure why that behavior was occurring.